# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/base/AccessibleOrProxy.cpp
# Commit: b20baffe6b38
# Full Hash: b20baffe6b38d4bc3109f64718f8be1f0bf0593f
# Author: James Teh <jteh@mozilla.com>
# Date: 2020-03-31 09:35:27
# Description:
#   Bug 1625396: AccessibleOrProxy::ChildAtPoint: Accessible::ChildAtPoint can return null, so don't try to check for a remote child doc in that case. r=yzen
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68739
# ==============================================================================

diff -r 187d340adf5e -r b20baffe6b38 accessible/base/AccessibleOrProxy.cpp
--- a/accessible/base/AccessibleOrProxy.cpp	Mon Mar 30 22:32:46 2020 +0000
+++ b/accessible/base/AccessibleOrProxy.cpp	Mon Mar 30 13:19:15 2020 +0000
@@ -47,8 +47,7 @@
     return childDoc->ChildAtPoint(aX, aY, aWhichChild);
   }
   AccessibleOrProxy target = AsAccessible()->ChildAtPoint(aX, aY, aWhichChild);
-  MOZ_ASSERT(target.IsAccessible());
-  if (aWhichChild == Accessible::eDirectChild) {
+  if (target.IsNull() || aWhichChild == Accessible::eDirectChild) {
     return target;
   }
   childDoc = target.RemoteChildDoc();
@@ -61,6 +60,7 @@
 }
 
 ProxyAccessible* AccessibleOrProxy::RemoteChildDoc() const {
+  MOZ_ASSERT(!IsNull());
   if (IsProxy()) {
     return nullptr;
   }
